app/screens/auth/SwitchNetwork.tsx

```diff
@@ -105,15 +105,25 @@ const SwitchNetwork = ({ history, location }: AuthRouterParams) => {
 
   const goNext = (genesisID: string | undefined) => {
     if (creatingWallet) {
-      if (isWalletOnly && genesisID?.length)
+      if (isWalletOnly && genesisID?.length) {
         return history.push(AuthPath.ConnectToAPI, {
-          redirect: AuthPath.CreateWallet,
+          redirect: AuthPath.ProtectWalletMnemonicStrength,
           genesisID,
           isWalletOnly,
           creatingWallet,
           mnemonic,
         });
-      return history.push(AuthPath.CreateWallet, {
+      }
+
+      if (mnemonic) {
+        return history.push(AuthPath.CreateWallet, {
+          genesisID,
+          isWalletOnly,
+          mnemonic,
+        });
+      }
```

Comment:
In which case this condition should work?
When User actually restoring the wallet?
Some explaining comments will be helpful